src/server.h

```diff
@@ -97,6 +97,13 @@ typedef struct redisObject robj;
 #define min(a, b) ((a) < (b) ? (a) : (b))
 #define max(a, b) ((a) > (b) ? (a) : (b))
 
+/* Offset of a member in a struct */
+#define member_offset(struct_name, member_name) ((size_t)&(((struct_name *)0)->member_name))
+
+/* Get the pointer of the outer struct from a member address */
+#define member2struct(struct_name, member_name, member_addr) \
```

Comment:
It look like it will generate a warning in FreeBSD, so we better rename it?
how about simply renaming it to `member_to_struct`? (i'm a bad name picker...)
```
  ./server.h:105:9: warning: 'member2struct' macro redefined [-Wmacro-redefined]
  #define member2struct(struct_name, member_name, member_addr) \
          ^
  /usr/include/sys/param.h:365:9: note: previous definition is here
  #define member2struct(s, m, x)                                          \
          ^
```

Reply:
just picking another name can result in another name clash.
IMO we can pick between these alternatives
1. add a `redis_` prefix
2. eliminate that macro and just do the math where we call it.
3. if the freebsd one does the same, we can use `#if !defined`

i guess option 1 is the preferred one.

Reply:
option 1 looks good `redis_member2struct`